### DISCUSSION TITLE
macOS Monterey cannot run dep command

### DISCUSSION THREAD

**Discussion by tlerbao** [deployphp/deployer#2983] [Help needed]
Error after running dep command：
<img width="350" alt="image" src="https://user-images.githubusercontent.com/20442232/156562639-b42df62d-1bae-4537-980a-b006d90fd0f3.png">

macOS Monterey 12.2 (21D49)。

I am able to run php commands：
<img width="675" alt="image" src="https://user-images.githubusercontent.com/20442232/156562828-c03ca432-58f3-4c91-a7f2-ff7179338fdc.png">

i can run laravel commands "php artsan xxx"
<img width="621" alt="image" src="https://user-images.githubusercontent.com/20442232/156565033-1d591acc-bf96-404f-bdff-e2f7711c3f9f.png">

My development environment is only running in docker

<img width="1322" alt="image" src="https://user-images.githubusercontent.com/20442232/156562978-df638209-1025-4dd1-be78-80825a40bfea.png">

**Does anyone know how to solve this problem? No projects have been deployed for two days! Many thanks!**

**Comment by antonmedv** [comment]
If you able to run php, just rub: php dep …

**Reply by tlerbao** [reply]
In big sur I can run the dep command.

And I installed deployer globally, the theory is that you can run the dep command

<img width="623" alt="image" src="https://user-images.githubusercontent.com/20442232/156570049-cca57630-036c-44e4-8128-abaae520ecf7.png">
You said running php dep doesn't work

<img width="504" alt="image" src="https://user-images.githubusercontent.com/20442232/156570097-48856723-39de-453d-ae90-21c18f14e3ed.png">

**Reply by antonmedv** [reply]
Yes, as you’re  running it inside container and container doesn’t have dep inside it.

**Reply by tlerbao** [reply]
I was able to run the dep command on the host during big sur without entering the container, but after upgrading monterey it didn't work for some reason: env: php: No such file or directory; so I came to ask for help.

**Reply by tlerbao** [reply]
<img width="499" alt="image" src="https://user-images.githubusercontent.com/20442232/156571002-a0e82cd7-8c38-401c-966b-f3a6c01873d5.png">

**Reply by tlerbao** [reply]
Why is my issue always deleted

**Reply by antonmedv** [reply]
Because it’s not a Deployer issue.

**Reply by tlerbao** [reply]
@antonmedv I didn't say it was the deployer's problem, but I couldn't confirm what caused it either. Indeed, at the time of big sur, my dep command is fully operational, outside the container, not inside the container. My development environment, docker, has not changed. I just upgraded the system to monterey, which has troubled me for several days and has not solved it; so I came here for help, hoping to get help.

**Reply by antonmedv** [reply]
Sure, discussions is the right place for this.

**Reply by tlerbao** [reply]
I observed that the first line of deployer/bin/dep is #!/usr/bin/env php, my php is running in the docker container, I do not have php installed on this machine, and the php command is run on the . It is implemented by adding relevant content in zshrc (as shown in the figure). I wonder if it is related to this #!/usr/bin/env php?

![](https://user-images.githubusercontent.com/20442232/156571002-a0e82cd7-8c38-401c-966b-f3a6c01873d5.png)

**Reply by antonmedv** [reply]
No it’s not. Your container does not have Deployer inside.

**Reply by tlerbao** [reply]
@antonmedv Maybe the English from google translate made my expression unclear:

At the time of macOS big sur, I only installed deployer on the local host, and php was running in the doker container. At that time, the dep command could be run on the local host, and there was no need to install deployer in the container, nor in the container. Run the dep command in it.

The only difference now is that I upgraded to macOS Monterey, but the dep command doesn't work anymore, I don't know why.

**Reply by antonmedv** [reply]
I don’t know 🤷‍♂️

**Reply by tlerbao** [reply]
@antonmedv After eliminating the possible factors, I reinstalled the operating system as Big sur, and used time machine to restore the full backup. Everything is back to normal, and the dep command can be executed.

I don't know why Monterey doesn't work. Maybe many of my friends haven't upgraded to this version. I believe that some people will encounter the same problem as me;

If the official finds the root cause of this problem in the future, I hope it will be written in the changelog so that I can update to Monterey again, thank you very much.
